repository_service_tuf_api/tasks.py

```diff
@@ -120,12 +122,18 @@ def get(task_id: str) -> Response:
         ``Response`` as BaseModel from pydantic
     """
     task = repository_metadata.AsyncResult(task_id)
+
+    task_state = task.state
+    task_result = task.result
+
     if isinstance(task.result, Exception):
         task_result = str(task.result)
-    else:
-        task_result = task.result
+    elif task_state == TaskState.SUCCESS and not task_result.get(
+        "status", False
+    ):
+        task_state = TaskState.ERRORED
 
     return Response(
-        data=TasksData(task_id=task_id, state=task.state, result=task_result),
+        data=TasksData(task_id=task_id, state=task_state, result=task_result),
```

Comment:
Shouldn't we also change the `task.state = task_state` as if it enters on line `131` then `task_state` will change, but `task.state` will remain `TASK.SUCCESS`?
This sounds that could be misleading.
@kairoaraujo any thoughts?

Reply:
Not sure if I understood.
We store `task.state` in `task_state`, and we no longer use the original object. (We do the same for `task_result`)